includes/admin/class-wc-subscriptions-admin.php

```diff
@@ -1387,6 +1390,68 @@ public static function filter_orders_table_by_related_orders( $query_vars ) {
 		return $query_vars;
 	}
 
+	/**
+	 * Filters the Admin orders and subscriptions table results in HPOS based on a list of IDs returned by a report query.
+	 *
+	 * @since 7.1.0
+	 *
+	 * @param array $clauses The query clause.
+	 *
+	 * @return array $clauses The query clause with additional `where` clause .
+	 */
+	public static function filter_orders_and_subscriptions_from_order_table( $clauses ) {
+		global $wpdb;
+
+		if ( ! ( isset( $_GET['page'] ) && 'wc-orders' === $_GET['page'] ) || ! isset( $_GET['_report'] ) ) {
```

Comment:
```suggestion
		if ( ! ( isset( $_GET['page'] ) && in_array( $_GET['page'], [ 'wc-orders', 'wc-orders--shop_subscription' ] ) ) || ! isset( $_GET['_report'] ) ) {
```

The way this logic is written at the moment this function won't work on the subscriptions list table page. 

If you click on any of these subscription specific ones, you'll be redirected to the subscriptions list table with no filtering applied. The suggestion I included above should fix that. 

<img width="243" alt="Screenshot 2024-04-29 at 12 06 34 PM" src="https://github.com/Automattic/woocommerce-subscriptions-core/assets/8490476/14733391-8d4a-46ea-8a90-0234607bf869">